File: C/Colleen.c
#include <stdio.h>
/*
	This program will print its own source when run.
*/
void hey(){}
int main(){
/*
	This program will print its own source when run.
*/
char *c="#include <stdio.h>%c/*%c%cThis program will print its own source when run.%c*/%cvoid hey(){}%cint main(){%c/*%c%cThis program will print its own source when run.%c*/%cchar *c=%c%s%c;printf(c, 10, 10, 9, 10, 10, 10, 10, 10, 9, 10, 10, 34, c, 34);hey();}";printf(c, 10, 10, 9, 10, 10, 10, 10, 10, 9, 10, 10, 34, c, 34);hey();}

File: PHP/Colleen.php
<?php
/*
    This program will print its own source when run.
*/
function main(){
/*
    This program will print its own source when run.
*/
$str = '<?php%c/*%c    This program will print its own source when run.%c*/%cfunction main(){%c/*%c    This program will print its own source when run.%c*/%c%cstr = %c%s%c;%cprintf($str, 10, 10, 10, 10, 10, 10, 10, 10, 36, 39, $str, 39, 10, 10, 10, 10, 10, 10);%c}%cfunction hey(){}%cmain();%c?>%c';
printf($str, 10, 10, 10, 10, 10, 10, 10, 10, 36, 39, $str, 39, 10, 10, 10, 10, 10, 10);
}
function hey(){}
main();
?>
